tests/test_oracle_implementation.py

```diff
@@ -73,20 +73,22 @@ async def test_update_oracle_controller_address(contract):
 
 @pytest.mark.asyncio
 async def test_submit_entries(contract, source, publisher):
-    entry = construct_entry(
+    entry = Entry(
         key="eth/usd", value=2, timestamp=1, source=source, publisher=publisher
     )
 
-    await contract.publish_entry(entry).invoke(caller_address=ORACLE_CONTROLLER_ADDRESS)
+    await contract.publish_entry(entry.serialize()).invoke(
```

Comment:
This is one of the drawbacks of the new setup, `contract.publish_entry(entry)` is cleaner and more intuitive than `contract.publish_entry(entry.serialize())`. Overall I think the net is positive, but just want to point out this aspect is a negative in dev experience (e.g. I would often forget the `.serialize` part).

Reply:
Though I guess this is for tests mainly, and devs would use our client which packages this/hides it away already

Reply:
we could just make publish_entry take a `Union[Entry, SerializedEntry]`

Reply:
I think this is more an argument for an abstraction on the contract interface, since this contract invocation is for the raw contract object, rather than an abstraction that can handle higher-kinded types